Exiled.Events/EventArgs/Map/Scp244SpawningEventArgs.cs

```diff
@@ -0,0 +1,62 @@
+// -----------------------------------------------------------------------
+// <copyright file="Scp244SpawningEventArgs.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.EventArgs.Map
+{
+    using API.Features;
+    using Exiled.API.Features.Pickups;
+    using Interfaces;
+
+    /// <summary>
+    ///     Contains all information up to spawning Scp244.
+    /// </summary>
+    public class Scp244SpawningEventArgs : IRoomEvent, IPickupEvent, IDeniableEvent
+    {
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="Scp244SpawningEventArgs" /> class.
+        /// </summary>
+        /// <param name="room">
+        ///     <inheritdoc cref="Room" />
+        /// </param>
+        /// <param name="pickup">
+        ///     <inheritdoc cref="Pickup" />
+        /// </param>
+        /// <param name="scp244Pickup">
+        ///     <inheritdoc cref="Scp244Pickup" />
+        /// </param>
+        /// <param name="isAllowed">
+        ///     <inheritdoc cref="IsAllowed" />
+        /// </param>
+        public Scp244SpawningEventArgs(Room room, Pickup pickup, Scp244Pickup scp244Pickup, bool isAllowed = true)
```

Comment:
What the reason to have both `Pickup` and `Scp244Pickup` (which inherits from `Pickup`)?

Reply:
Isn't it better to give the developer the choice of what to use?
If he is interested in Pickup itself, let him use it.
If he is interested in Scp244Pickup, let him work with it.
For some reason there is a division by roles, for example: Scp106Role

Reply:
You can get it like `Pickup = scp244Pickup`. And pickup and scp244Pickup is same anyways

Reply:
Hmmm, Okay

Reply:
Or already inside contructor cast to Scp244Pickup

Reply:
I think you don't really understandede me. Property should be but you remove constructor argument need to be removed

Reply:
I still don't understand you

Reply:
You need to remove only construction argument